Nasal/v-speed.nas

```diff
@@ -85,28 +82,55 @@ var vspeed = {
 			# whiteBugSpeed.setValue(whiteBug);
 		} elsif (mode == 6) {
 			vSpeedSelectorModeText.setValue("SET");
-			if (airgnd == 1) {
-				if (me.v1 != nil) {
-					v1Speed.setValue(me.v1);
-				}
-				if (me.vr != nil) {
-					vrSpeed.setValue(me.vr);
-				}
-			} elsif (airgnd == 0) {
-				if (me.vref != nil) {
-					vrefSpeed.setValue(me.vref);
-				}
+			if (me.v1 != nil) {
+				v1Speed.setValue(me.v1);
+			}
+			if (me.vr != nil) {
+				vrSpeed.setValue(me.vr);
+			}
+			if (me.vref != nil) {
+				vrefSpeed.setValue(me.vref);
 			}
 			gWeight.setValue(me.wt);
 			setFlag.setBoolValue(1);
 		}
     },
+	
+	speedKnob: func() {
+		mode = vSpeedSelectorMode.getValue();
+		airgnd = airground.getValue();
+		bug = vSpeedBug.getValue();
+		if (mode == 0 or mode == 6) {return;}
+		
+		if (mode == 1 and airgnd == 1) {
+			me.v1 = bug;
+			vSpeedSelectorPFDText.setValue(me.v1);
+		} elsif (mode == 2 and airgnd == 1) {
+			me.vr = bug;
+			vSpeedSelectorPFDText.setValue(me.vr);
+		} elsif (mode == 3) {
+			wtRaw = bug;
+			me.wt = wtRaw * 500;
+			vSpeedSelectorPFDText.setValue(me.wt);
+		} elsif (mode == 4 and airgnd == 0) {
+			me.vref = bug;
+			vSpeedSelectorPFDText.setValue(me.vref);
+		}
+	},
 };
 
 ###############################################################################
-# Init and start loop
+# Listeners
 ###############################################################################
+setlistener("/controls/fmc/v-speed-mode", func {
+	vspeed.modeKnob();
+}, 0, 0);
 
-var vspeedLoop = maketimer(0.05, func {vspeed.update();});
+setlistener("/controls/fmc/v-speed-bug", func {
+	vspeed.speedKnob();
+}, 0, 0);
 
```

Comment:
Hey, this also works, but actually I thought you would just use 'update' as the same listener for both properties. Anyway, this is also absolutely fine.